builder/builder.go

```
@@ -58,81 +58,115 @@ func getAuthData(filename string) (*dockerAuthData, error) {
 
-	buildContext := context.Background()
+	backgroundContext := context.Background()
 	dockerClient, err := client.NewEnvClient()
 	if err != nil {
 		log.Fatalf("error in creating a docker client: %v", err)
+		return err
+	}
+
+	// Create a build context tar for the image.
+	dockerFilepath := imgBuilder.dockerFilename
+
+	dockerBuildContext, err := getDockerBuildContextTar(dockerFilepath)
+	if err != nil {
+		log.Fatalf("error in creating a docker client: %v", err)
+		return err
 	}
 
-	// Create a tar for build context.
-	tarBuffer := new(bytes.Buffer)
-	writer := tar.NewWriter(tarBuffer)
+	tagName := getTagName(imgBuilder, assignmentEnv)
+
+	response, err := dockerClient.ImageBuild(
+		backgroundContext,
+		dockerBuildContext,
+		types.ImageBuildOptions{
+			Dockerfile: dockerFilepath,
+			Tags:       []string{tagName}})
+	if err != nil {
+		log.Fatalf("unable to build docker image: %v", err)
+		return err
+	}
 	defer func() {
-		err = writer.Close()
+		err = response.Body.Close()
 		if err != nil {
 			log.Fatal(err)
 			return
 		}
 	}()
+	_, err = io.Copy(os.Stdout, response.Body)
+	if err != nil {
+		log.Fatalf("unable to read image build response: %v", err)
+		return err
+	}
 
-	dockerFilepath := imgBuilder.dockerFilename
+	return nil
+}
+
+// Get Tag name for building image.
+func getTagName(imgBuilder ImageBuilder, assignmentEnv bool) string {
+	var tagName string
+	if assignmentEnv {
+		tagName = fmt.Sprintf("%s/%s", imgBuilder.authData.Username, imgBuilder.LanguageImageFormat)
+	} else {
+		tagName = fmt.Sprintf("%s/%s:%s", imgBuilder.authData.Username,
+			imgBuilder.authData.Repository, imgBuilder.authData.Version)
+	}
+
+	return tagName
+}
+
+// Get Docker Build Context Tar Reader for building image.
+func getDockerBuildContextTar(dockerFilepath string) (*os.File, error) {
 	dockerFileReader, err := os.Open(dockerFilepath)
 	if err != nil {
 		log.Fatalf(" unable to open dockerfile: %v", err)
```

why not return the error here like done in other places?

---

Fixed